docs: adds callouts linking to Gatsby Cloud

gillkyle opened the pull request:
## Description

This pull request adds a `CloudCallout` component and links it in docs that serve as logical stepping stones into things Gatsby Cloud helps solve.

![image](https://user-images.githubusercontent.com/21114044/79801243-00dc1380-831b-11ea-94da-b51431d89067.png)

There is also an additional call to action added on the homepage that keeps the primary action linking users to the docs. For those already using Gatsby and looking for Cloud this creates a more bi-directional flow between Gatsby Cloud and the docs.

![image](https://user-images.githubusercontent.com/21114044/79801224-f4f05180-831a-11ea-9cb3-476c78aeef48.png)


## Related Issues

_None_

laurieontech commented on docs/contributing/docs-and-blog-components.md:
```diff
@@ -124,6 +124,38 @@ Rendered, the component looks like this:
+
+#### Usage
+
+The Cloud Callout component takes one optional props, and prepends the children it wraps to the general call to action:
```
```suggestion
The Cloud Callout component takes one optional prop, and prepends the children it wraps to the general call to action:
```

gillkyle commented:
Thanks @laurieontech! I didn't check dark mode, those were great points. I made the text match the size of the rest of the page, used a purple that has more contrast, updated that svg circle to remove the fill since it really stands out in dark mode. 

I also fixed that mistake with the plural props 🙂

laurieontech commented:
Looks great! The only slightly weird thing is that the `Gatsby Cloud` link is no longer emphasized in light mode. Intentional?
<img width="659" alt="Screen Shot 2020-04-21 at 4 35 10 PM" src="https://user-images.githubusercontent.com/15000607/79911276-19692e00-83ee-11ea-8330-6972ccec2518.png">

gillkyle commented:
That's a very valid point, I actually don't see any huge reason to make all that other text purple. 

I'll updated it so just the link text shows as purple.

![image](https://user-images.githubusercontent.com/21114044/79923872-b1b5e180-83f3-11ea-8de7-815ec69e1904.png)

![image](https://user-images.githubusercontent.com/21114044/79923881-b7abc280-83f3-11ea-842d-6e856c35c0ab.png)

laurieontech commented:
Looks good! Sorry for being so nitpicky.
